myserver.cpp

```diff
@@ -0,0 +1,684 @@
+#include "myserver.h"
+#include "IDataBase.cpp"
+#include "IDataBase.h"
+#include "DBErrors.h"
+#include <iostream>
+#include <vector>
+#include <string>
+
+myserver::myserver(){}
+
+myserver::~myserver(){}
+
+void myserver::startServer(){
+    if(this->listen(QHostAddress::Any,5555)){
+        qDebug()<<"Listening";
+    }
+    else{
+        qDebug()<<"Not listening";
+    }
+}
+
+void myserver::incomingConnection(qintptr pisocketDescriptor){
+    socket = new QTcpSocket(this);
+    socket->setSocketDescriptor(pisocketDescriptor);
+
+    connect(socket, SIGNAL(readyRead()),this,SLOT(sockReady()));
+    connect(socket, SIGNAL(disconnected()),this,SLOT(sockDisc()));
+
+    qDebug()<<pisocketDescriptor<<"Client connected";
+
+    socket->write("{\"type\":\"connect\",\"status\":\"yes\"}");
+    qDebug()<<"Send client connect status - YES";
+}
+
+void myserver::sockReady(){
+    Data = socket->readAll();
+    qDebug()<<"Data: "<<Data;
+    std::vector <std::string> msg;
+
+    nlohmann::json j;
+    j = nlohmann::json::parse(QString(Data).toStdString());
+
+    for (nlohmann::json::iterator it = j.begin(); it != j.end(); ++it) {
+            msg.push_back(it.value());
+    }
+    QJsonDocument document;
+    QByteArray bytes;
+
+        if(msg[0]=="AddUser"){
+            int iUserId;
+            int iRtCode = IDataBase::AddUser(msg[1], msg[2], iUserId);
+            QJsonObject AddUser;
+            switch(iRtCode){
+            case SUCCESS:
+                AddUser.insert("RequestType", "AddUser");
+                AddUser.insert("SUCCESS", iUserId);
+                document.setObject(AddUser);
+                bytes = document.toJson( QJsonDocument::Indented );
+                socket->write(bytes);
+                socket->waitForBytesWritten(500);
+                break;
+            case INVALID_INPUT_DATA:
+                res = "INVALID_INPUT_DATA";
+                socket->write(res);
+                socket->waitForBytesWritten(500);
```

Comment:
or this. It is almost the same as in next case.

Reply:
Fixed it